Cannot use the Odoo SMTP with another mail server (#2075)
odoo/documentation: content/applications/general/email_communication/email_servers.rst

====================================================
Send and receive emails in Odoo with an email server
====================================================

Odoo Online or Odoo.sh users
============================

> Nothing needs to be done! **Odoo sets up its own mail servers for the database.** Outgoing and
> incoming emails work out-of-the-box!
> 
> Unless an external mail server is required to send large batches of mass mailing, simply enjoy the
> new Odoo database.

Reviewer: Suggestions around sentence structure to make this section clearer to understand (optional).
```suggestion
Since **Odoo sets up its own mail servers for the database**, outgoing and incoming emails already
work out-of-the-box. So for **Odoo Online** and **Odoo.sh** customers, nothing needs to be done!

Unless an external mail server is required to send large batches of mass mailing, simply use the
standard online Odoo database normally since it has already been preconfigured for email.
```

Scope of this documentation
===========================

This document is **mainly dedicated to Odoo on-premise users** who don't benefit from an
out-of-the-box solution to send and receive emails in Odoo, unlike `Odoo Online <https://www.odoo.
com/trial>`_ & `Odoo.sh <https://www.odoo.sh>`_.

.. warning::
   If no one in the company is used to managing email servers, Odoo Online and Odoo.sh are strongly
   recommended. Those Odoo hosting types' email systems work instantly and are monitored by
   professionals. Nevertheless, a company can use their own email server if they want to manage the
   email server's reputation themselves.

> The following sections contain information on how to integrate an external email server with Odoo.

Reviewer: Would move this sentence above the `.. warning::` tag since it's a continuation of that first paragraph.
Would also add "below" to indicate proceeding information as user scrolls down.
```suggestion
The following sections below contain information on how to integrate an external email server with
Odoo.
```

enable the :guilabel:`External Email Servers` option. Then, click :guilabel:`Save`. Next, click
:guilabel:`Outgoing Email Servers` and click :guilabel:`Create` to create a new outgoing mail
server record in Odoo. Reference the SMTP data of the external email server. Once all the
information has been filled out, click :guilabel:`Test Connection`.

Here is a typical configuration for a G Suite server.

.. image:: email_servers/outgoing-server.png
   :align: center
   :alt: The typical G Suite configuration on Odoo.

Then, go back to :menuselection:`Settings --> General Settings --> Discuss` and set the email
domain name in the :guilabel:`Alias Domain` name. Finally, click :guilabel:`Save` to finish setting
up the email server.

.. note::
   If an `[AUTHENTICATION FAILED] Invalid credentials (Failure)` warning appears when
   :guilabel:`Test Connection` is clicked for a Gmail address, activate the :guilabel:`Less secure
   app access` option. A direct link can be `accessed here <https://myaccount.google.com/
   lesssecureapps?pli=1>`_.

   In addition to that, enable the :guilabel:`IMAP setting` on the Gmail account.

> Can an Office 365 server be used
> --------------------------------

Reviewer: This heading is strange since it's a question but there's no question mark.
Suggestion below to match the format of the G Suite heading that immediately follows.
```suggestion
Use an Office 365 server
------------------------
```

An Office 365 server can be used if the database's hosting type is **Odoo on-premise**. Office 365
SMTP relays are not compatible with Odoo Online or Odoo.sh unless Odoo is configured to
:ref:`force the outgoing "From" address <email_communication/default_from>`.

Please refer to `Microsoft's documentation <https://support.office.com/en-us/article/How-to-set-up-
a-multifunction-device-or-application-to-send-email-using-Office-365-69f58e99-c550-4274-ad18-
c805d654b4c4>`_ to configure an SMTP relay for the Odoo database's IP address.

Use a G Suite server
--------------------

A G Suite server can be used for any Odoo hosting type. To do so, set up the SMTP relay service.
The configuration steps are explained in `Google documentation <https://support.google.com
/a/answer/2956491?hl=en>`__.

Restriction
-----------

Please note that port 25 is blocked for security reasons on the SaaS and Odoo.sh platforms. Try

> using 465, 587, or 2525.

Reviewer: Context.
```suggestion
using ports 465, 587, or 2525 instead.
```

.. _email_communication/default_from:

Use a default "From" email address
----------------------------------

Sometimes, an email's "From" (outgoing) address can belong to a different domain, and that can be a
problem.

For example, if a customer with the email address *mary\@customer.example.com* responds to a
message, Odoo will try to redistribute that same email to the other subscribers in the thread.
However, if the domain *customer.example.com* forbids that kind of usage for security, the email
that Odoo is trying to redistribute would get rejected by some recipients' email servers.

> To avoid that problem, make sure all emails use a "From" address from the authorized domain.

Reviewer: Context.
```suggestion
To avoid that problem, make sure all emails use a "From" address from the same authorized domain.
```

  domain name instead of the external email server is recommended. That way, incoming messages can
  be received without delay. Odoo Online only fetches incoming messages of external servers once
  every hour. Redirections for all email addresses should be set to Odoo's domain name in the email
  server (e.g. *catchall\@mydomain.ext* to *catchall\@mycompany.odoo.com*).

All the aliases are customizable in Odoo. Object aliases can be edited from their respective
configuration view.

.. tip:: To edit catchall and bounce aliases, first activate the :ref:`developer mode
   <developer-mode>`. Then, go to :menuselection:`Settings --> Technical --> Parameters --> System
   Parameters` to customize the aliases (*mail.catchall.alias* & *mail.bounce.alias*).

   .. image:: email_servers/system-parameters.png
      :align: center
      :alt: System parameters with catchall configuration in Odoo.

By default, inbound messages are fetched every 5 minutes in Odoo on-premise.

.. note::
   This value can be changed in :ref:`developer mode <developer-mode>`. Go to
   :menuselection:`Settings --> Technical --> Automation --> Scheduled Actions` and look for

>    *Mail: Fetchmail Service*.

Reviewer: Indicating to look for a UI field therefore change to `:guilabel:`
```suggestion
   :guilabel:`Mail: Fetchmail Service`.
```

Use a default "From" email address
----------------------------------

Sometimes, an email's "From" (outgoing) address can belong to a different domain, and that can be a
problem.

For example, if a customer with the email address *mary\@customer.example.com* responds to a
message, Odoo will try to redistribute that same email to the other subscribers in the thread.
However, if the domain *customer.example.com* forbids that kind of usage for security, the email
that Odoo is trying to redistribute would get rejected by some recipients' email servers.

To avoid that problem, make sure all emails use a "From" address from the same authorized domain.

If the MTA supports `SRS (Sender Rewriting Scheme) <https://en.wikipedia.org/wiki/Sender_Rewriting
_Scheme>`_, SRS can be enabled to handle these situations. However, that is more complex and
requires more technical knowledge that is not meant to be covered by this documentation.

Instead, Odoo can be configured to do something similar by itself:

#. Set the :guilabel:`Alias Domain` name in the :menuselection:`Settings --> General Settings
   --> Discuss`.

   .. image:: email_servers/alias-domain.png

>       :align: center

Reviewer 2: I'd align the image to the left to clearly show it's part of the bullet point.
```suggestion
```

Reviewer 3: Hi @AntoineVDV,
When aligned left, the following bullets wrap the image. Is there a way to avoid this?

![Screen Shot 2022-09-19 at 9 41 00 AM](https://user-images.githubusercontent.com/36018073/191031738-9402a0e1-7ce2-4cda-b0ef-c671954233bb.png)

Reviewer 2: If you surround the image with blank lines it should work. In rst, when something doesn't work, add blank lines!

Reviewer 3: I tried multiple variations of adding blank lines and it still wrapped around the image 🫤 (I ran `make clean` then `make html` every time I made a change). I did find a work around by adding the `:width: 100%` attribute, but I'm not sure if this is good practice - if not I can remove the attribute 👍

  Discuss`.

* If the database's hosting type is Odoo on-premise, create an :guilabel:`Incoming Mail Server` in
  Odoo for each alias. This can be done from the General Settings as well. Fill out the form
  according to the email provider's settings. Leave the :guilabel:`Actions to Perform on Incoming
  Mails` field blank. Once all the information has been filled out, click on :guilabel:`TEST &
  CONFIRM`.

  .. image:: email_servers/incoming-server.png
     :align: center
     :alt: Incoming mail server configuration on Odoo.

* If the database's hosting type is Odoo Online or Odoo.sh, redirecting incoming messages to Odoo's
  domain name instead of the external email server is recommended. That way, incoming messages can
  be received without delay. Odoo Online only fetches incoming messages of external servers once
  every hour. Redirections for all email addresses should be set to Odoo's domain name in the email
  server (e.g. *catchall\@mydomain.ext* to *catchall\@mycompany.odoo.com*).

All the aliases are customizable in Odoo. Object aliases can be edited from their respective
configuration view.

> .. tip:: To edit catchall and bounce aliases, first activate the :ref:`developer mode
>    <developer-mode>`. Then, go to :menuselection:`Settings --> Technical --> Parameters --> System
>    Parameters` to customize the aliases (*mail.catchall.alias* & *mail.bounce.alias*).

Reviewer 2: Nitpicking (i.e. no need to change that here, but take into account for the next PRs): it's better to start the content of tips/notes on the next line, otherwise it's interpreted as the tip/note's title, which is not supported.
```suggestion
.. tip::
   To edit catchall and bounce aliases, first activate the :ref:`developer mode
   <developer-mode>`. Then, go to :menuselection:`Settings --> Technical --> Parameters --> System
   Parameters` to customize the aliases (*mail.catchall.alias* & *mail.bounce.alias*).
```

   .. image:: email_servers/system-parameters.png
      :align: center
      :alt: System parameters with catchall configuration in Odoo.

By default, inbound messages are fetched every 5 minutes in Odoo on-premise.

> .. note::

Reviewer 2: Tip?

Use a default "From" email address
----------------------------------

Sometimes, an email's "From" (outgoing) address can belong to a different domain, and that can be a
problem.

For example, if a customer with the email address *mary\@customer.example.com* responds to a
message, Odoo will try to redistribute that same email to the other subscribers in the thread.
However, if the domain *customer.example.com* forbids that kind of usage for security, the email
that Odoo is trying to redistribute would get rejected by some recipients' email servers.

To avoid that problem, make sure all emails use a "From" address from the same authorized domain.

If the MTA supports `SRS (Sender Rewriting Scheme) <https://en.wikipedia.org/wiki/Sender_Rewriting
_Scheme>`_, SRS can be enabled to handle these situations. However, that is more complex and
requires more technical knowledge that is not meant to be covered by this documentation.

Instead, Odoo can be configured to do something similar by itself:

#. Set the :guilabel:`Alias Domain` name in the :menuselection:`Settings --> General Settings
   --> Discuss`.

   .. image:: email_servers/alias-domain.png

>       :align: left
>       :width: 100%

Reviewer 2: Sorry, I didn't make myself clear, to align left, you must drop the align center, and that's it. The `align: left` option is not supported. :sweat_smile: 
```suggestion
```
-->
```rst
.. image:: email_servers/alias-domain.png
   :alt: Setting the domain alias configuration on Odoo.
```

> Send and Receive Emails in Odoo with an Email Server

Reviewer 4: Replace entire header with `Send and receive emails in Odoo with an email server`

As a system admin, go to :menuselection:`Settings --> General Settings` and check
:guilabel:`External Email Servers`. Then, click :guilabel:`Outgoing Mail Servers` to create one and
reference the SMTP data of your email server. Once all the information has been filled out, click
on :guilabel:`Test Connection`.

Here is a typical configuration for a G Suite server.

.. image:: email_servers/outgoing-server.png
   :align: center
   :alt: The typical G Suite configuration on Odoo.

Then set your email domain name in the General Settings.

.. note::
   If you get a `[AUTHENTICATIONFAILED] Invalid credentials (Failure)` warning when you
   :guilabel:`Test Connection` on a Gmail address, activate the :guilabel:`Less secure app access`
   option. A direct link can be `accessed here <https://myaccount.google.com/
   lesssecureapps?pli=1>`_.

   In addition to that, enable the :guilabel:`IMAP setting` on your Gmail account.

Can I use an Office 365 server
------------------------------

You can use an Office 365 server if you run Odoo on-premise. Office 365 SMTP relays are not
compatible with Odoo Online unless you configure Odoo to :ref:`force the outgoing "From" address

> <email_communication/default_from>`.

Reviewer 5: ```suggestion
address <email_communication/default_from>`.
```

Please refer to `Microsoft's documentation <https://support.office.com/en-us/article/How-to-set-up-
a-multifunction-device-or-application-to-send-email-using-Office-365-69f58e99-c550-4274-ad18-
c805d654b4c4>`_ to configure a SMTP relay for your Odoo's IP address.

How to use a G Suite server
---------------------------

You can use an G Suite server for any Odoo hosting type. To do so you need to setup the SMTP relay
service. The configuration steps are explained in `Google documentation <https://support.google.com
/a/answer/2956491?hl=en>`__.

Restriction
-----------

Please note that port 25 is blocked for security reasons on our SaaS and Odoo.sh platform. Try
using 465, 587, or 2525.

> .. _email_communication/default_from:

Reviewer 4: Not sure what this is supposed to be?

Reviewer 3: @mivu-odoo this is a [custom anchor](https://www.odoo.com/documentation/15.0/contributing/documentation/rst_cheat_sheet.html#custom-anchors) for internal page linking 😁

Reviewer 4: Oooh got it! I saw a couple more sprinkled throughout the doc, so I figured it was there for a reason. Thank you for teaching me something new 😸

Use a default "From" email address
----------------------------------

Sometimes, an email's "From" (outgoing) address can belong to a different domain, and that can be a
problem.

For example, if a customer with address *mary\@customer.example.com* responds to a message, Odoo
will try to redistribute that same email to other subscribers in the thread. But if the domain
*customer.example.com* forbids that kind of usage for security, the Odoo's redistributed email

> would get rejected by some recipients' mail servers.

Reviewer 4: Replace the entire paragraph `For example...some recipients' mail servers.` with:

`For example, if a customer with the email address *mary\@customer.example.com* responds to a message, Odoo will try to redistribute that same email to the other subscribers in the thread. However, if the domain *customer.example.com* forbids that kind of usage for security, the email that Odoo is trying to redistribute would get rejected by some recipients' email servers.`

To avoid those kind of problems, you should make sure all emails use a "From" address from your

> authorized domain.

Reviewer 4: Replace the entire sentence `To avoid those...authorized domain.` with:

`To avoid that problem, make sure all emails use a "From" address from the authorized domain.`

If your MTA supports `SRS (Sender Rewriting Scheme) <https://en.wikipedia.org/wiki/Sender_Rewriting

> _Scheme>`_, you can enable it to handle these situations. However, that is more complex and

Reviewer 4: - Replace `your MTA` with `the MTA`
- Replace `you can enable it` with `SRS can be enabled`

requires more technical knowledge that is not meant to be covered by this documentation.

> Instead, you can also configure Odoo to do something similar by itself:

Reviewer 4: Replace `you can also configure Odoo to do something similar by itself` with `Odoo can be configured to do something similar by itself`

> #. Set your domain name in the :guilabel:`General Settings`.

Reviewer 4: - Replace `your domain name` with `` the :guilabel:`Alias Domain` name ``
- Replace `` the :guilabel:`General Settings` `` with `` :menuselection:`Settings --> General Settings --> Discuss` ``

   .. image:: email_servers/alias-domain.png
      :align: center
      :alt: Setting the domain alias configuration on Odoo.

#. In developer mode, go to :menuselection:`Settings --> Technical --> Parameters --> System

>    Parameters`.

Reviewer 4: - Add a line before `` #. In developer mode...Parameters`. `` that says `` #. Turn on :doc:`developer mode </applications/general/developer_mode>` ``
- Delete `In developer mode`
- Capitalize the `g` in `go to`

> #. Add one system parameter from these:

Reviewer 4: Replace `these` with `the following list`

>    * If you want *all* your outgoing messages to use the same "From" address, use the key

Reviewer 4: Replace `If you want *all* your outgoing messages to use the same "From" address` with

`To use the same "From" address for *all* outgoing messages`

     `mail.force.smtp.from` and set that address as value (such as `outgoing@mycompany.example.com`
     ).

>    * If you want to keep the original "From" address for emails that use your same domain, but change

Reviewer 4: Replace `If you want to keep the original "From" address for emails that use your same domain` with `To keep the original "From" address for emails that use the same domain`

> 

Reviewer 4: I couldn't comment on the line that had the header title `How to manage inbound messages`, but please

- Delete `How to`
- Capitalize the `m` in `manage`

Odoo relies on generic email aliases to fetch incoming messages.

* **Reply messages** of messages sent from Odoo are routed to their original discussion thread (and
  to the inbox of all its followers) by the catchall alias (**catchall@**).
* **Bounced messages** are routed to **bounce@** in order to track them in Odoo. This is especially
  used in `Odoo Email Marketing <https://www.odoo.com/page/email-marketing>`__ to opt-out invalid
  recipients.
* **Original messages**: Several business objects have their own alias to create new records in
  Odoo from incoming emails:

  * Sales Channel (to create Leads or Opportunities in `Odoo CRM <https://www.odoo.com/page/crm>`__),
  * Support Channel (to create Tickets in `Odoo Helpdesk <https://www.odoo.com/page/helpdesk>`__),
  * Projects (to create new Tasks in `Odoo Project <https://www.odoo.com/page
    /project-management>`__),
  * Job Positions (to create Applicants in `Odoo Recruitment <https://www.odoo.com/page
    /recruitment>`__),
  * etc.

> Depending on your mail server, there might be several methods to fetch emails. The easiest and most

Reviewer 4: Replace `your` with `the`

> recommended method is to manage one email address per Odoo alias in your mail server.

Reviewer 4: Replace `your` with `the`

> * Create the corresponding email addresses in your mail server

Reviewer 4: Replace `your` with `the

  (catchall@, bounce@, sales@, etc.).

> * Set your domain name in the General Settings.

Reviewer 4: Replace `your domain name in the General Settings` with `` the :guilabel:`Alias Domain` name in :menuselection:`Settings --> General Settings --> Discuss` ``

  .. image:: email_servers/alias-domain.png
      :align: center

>       :alt: Setting the domain alias configuration on Odoo.

Reviewer 4: Is this really necessary here? We've already seen this image and mentioned this setting multiple times in the previous sections. I would just delete this image markup and its tags (but NOT the actual image file).

* If you use Odoo on-premise, create an :guilabel:`Incoming Mail Server` in Odoo for each alias.
  You can do it from the General Settings as well. Fill out the form according to your email
  provider's settings. Leave the :guilabel:`Actions to Perform on Incoming Mails` blank. Once all

>   the information has been filled out, click on :guilabel:`TEST & CONFIRM`.

Reviewer 4: - Replace `If you use Odoo on-premise` with `If the database's hosting type is Odoo on-premise`
- Replace `You can do it` with `This can be done`
- Replace `your email provider's` with `the email provider's`

  .. image:: email_servers/incoming-server.png
    :align: center
    :alt: Incoming mail server configuration on Odoo.

* If you use Odoo Online or Odoo.sh, We do recommend to redirect incoming messages to Odoo's domain
  name rather than exclusively use your own email server. That way you will receive incoming
  messages without delay. Indeed, Odoo Online is fetching incoming messages of external servers
  once per hour only. You should set redirections for all the email addresses to Odoo's domain name

>   in your email server (e.g. *catchall\@mydomain.ext* to *catchall\@mycompany.odoo.com*).

Reviewer 4: - Replace the two sentences `If you use Odoo Online or Odoo.sh...your own email server. That way...without delay.` with `If the database's hosting type is Odoo Online or Odoo.sh, redirecting incoming messages to Odoo's domain name instead of the external email server is recommended. That way incoming messages can be received without delay`
- Replace `You should set redirections for all the email addresses to Odoo's domain name in your email server` with `Redirections for all email addresses should be set to Odoo's domain name in the email server`

> .. tip:: All the aliases are customizable in Odoo. Object aliases can be edited from their

Reviewer 4: I think `All the aliases...` needs to be on its own line, not the same line as the `.. tip::`

>    respective configuration view. To edit catchall and bounce aliases, you first need to activate

Reviewer 4: - Delete `you` and `need to`

   *mail.bounce.alias*).

   .. image:: email_servers/system-parameters.png
      :align: center
      :alt: System parameters with catchall configuration in Odoo.

.. note:: By default inbound messages are fetched every 5 minutes in Odoo on-premise. You can
   change this value in :ref:`developer mode <developer-mode>`. Go to :menuselection:`Settings -->
   Technical --> Automation --> Scheduled Actions` and look for *Mail: Fetchmail Service*.

Set up different dedicated servers for transactional and mass mails
===================================================================

In Odoo a separate Mail Transfer Agent (MTA) server can be used for transactional e-mails and mass
mailings. Example: Use Postmark or SendinBlue for transactional e-mails, and Amazon SES, Mailgun or
Sendgrid for mass mailings.

.. note::
   A default outgoing email server is already configured. Do not create an alternative one unless a
   specific external outgoing email server is needed for technical reasons.

To do this, first activate the :ref:`developer mode <developer-mode>`, and then go to
:menuselection:`Settings --> Technical --> Outgoing` e-mail servers. There, create two e-mail MTA

> server settings; one for the transactional e-mails and one for the mass mailing server. Make sure

Reviewer 4: Replace `e-mails` with `emails` for consistency with the rest of the doc

(two instances of `e-mails` here)

Set up different dedicated servers for transactional and mass mails
===================================================================

> You can use a separate Mail Transfer Agent (MTA) servers for transactional e-mails and mass mailings.

Reviewer 5: Hi Jonathan, thanks for submitting this change. I am just going to change some of the wording, everything else looks great. 

``In Odoo a separate Mail Transfer Agent (MTA) server can be used for transactional e-mails and mass mailings.``

> Example: use Postmark for transactional e-mails, and Amazon SES

Reviewer 5: ``Example: Use Postmark or SendinBlue for transactional e-mails, and Amazon SES, Mailgun or Sendgrid for mass mailings.``

Set up different dedicated servers for transactional and mass mails
===================================================================

In Odoo a separate Mail Transfer Agent (MTA) server can be used for transactional e-mails and mass mailings.
Example: Use Postmark or SendinBlue for transactional e-mails, and Amazon SES, Mailgun or Sendgrid for mass mailings.

.. note::
   A default outgoing email server is already configured. You should not create an alternative one

>    unless you want to use a specific external outgoing email server for technical reasons.

Reviewer 4: Replace `You should not create an alternative one unless you want to use a specific external outgoing email server for technical reasons` with `Do not create an alternative one unless a specific external outgoing email server is needed for technical reasons.`

> To do this, you should first activate the :ref:`developer mode <developer-mode>` and then go to

Reviewer 4: - Delete `you should`
- Add a comma after `` :ref:`developer mode <developer-mode>` ``

> :menuselection:`Settings --> Technical --> Outgoing` e-mail servers. There you have to create two

Reviewer 4: - Add a comma after `There`
- Delete `you have to`

e-mail MTA server settings; one for the transactional e-mails and one for the mass mailing server. 
Make sure to give priority to the transactional server over the mass mailing server by providing 
a lower priority number for the transactional MTA server.

Now, go to :menuselection:`Email Marketing --> Settings` and enable *Dedicated Server*.
With these settings, Odoo uses the server with the lower priority for transactional emails, and the
server here selected for mass mails.

> Note that in this case, you have to set your domain's Sender Policy Framework (SPF) records to

Reviewer 4: Replace `you have to set your domain's Sender Policy Framework (SPF) records` with `the domain's Sender Policy Framework (SPF) records must be set `

To do this, you should first activate the :ref:`developer mode <developer-mode>` and then go to
:menuselection:`Settings --> Technical --> Outgoing` e-mail servers. There you have to create two

> e-mail MTA server settings. One for transactional e-mails and one for mass mail servers. 

Reviewer 5: ``e-mail MTA server settings; one for the transactional e-mails and one for the mass mailing server. Make sure to give priority to the transactional server over the mass mailing server by providing a lower priority number for the transactional MTA server.``
